dev/source-release.sh

```diff
@@ -66,7 +66,7 @@ tarball=$tag.tar.gz
 
 # be conservative and use the release hash, even though git produces the same
 # archive (identical hashes) using the scm tag
-git archive $release_hash --prefix $tag/ -o $tarball .baseline api arrow bundled-guava common core data dev flink gradle gradlew hive mr orc parquet pig spark spark2 spark-runtime spark3 spark3-runtime LICENSE NOTICE README.md build.gradle baseline.gradle deploy.gradle tasks.gradle jmh.gradle gradle.properties settings.gradle versions.lock versions.props version.txt
+git archive $release_hash --worktree-attributes --prefix $tag/ -o $tarball HEAD
```

Comment:
The `--worktree-attributes` is what tells `git archive` to check (all of) the `.gitattribute` files. We're keeping it simple and using one top level one, as it suit our needs.

I tested this and the resulting tarball had none of the files we expected to be ignored in its contents and had all of the files we expected to be there (when comparing the output sorted list from the current script to this script via the `uniq` shell command.